Add well known secret masking (#517)
spacelift-io/user-documentation: docs/concepts/stack/stack-settings.md

    Use with caution.

### Enable well known secret masking

This setting determines if secret patterns will be automatically redacted from logs. If enabled, the following secrets will be masked from logs:

- AWS Access Key Id
- GitHub PAT
- GitHub Fine-Grained PAT
- GitHub App Token
- GitHub Refresh Token
- GitHub OAuth Access Token
- Slack Token
- PGP Private Key
- RSA Private Key

> - PKCS#8 Private Key

Reviewer: I this is slightly inaccurate. We match the exchange format here (base64 encoded PEM), we don't really care about the key format.
Maybe we can just write `PEM blocks with BEGIN PRIVATE KEY header` or something like that?

Author: @eliecharra it makes sense

    Use with caution.

> ### Enable Well Known Masking

Reviewer 2: ```suggestion
### Enable well known secret masking
```

I think this should use Sentence case

> Indicates whether secret patterns will be automatically redacted from logs.
> If this is enabled, next secrets will be masked from logs:

Reviewer 2: ```suggestion
This setting determines if secret patterns will be automatically redacted from logs. If enabled, the following secrets will be masked from logs:
```

- AWS Access Key Id
- GitHub PAT
- GitHub Fine-Grained PAT
- GitHub App Token
- GitHub Refresh Token
- GitHub OAuth Access Token
- Slack Token
- PGP Private Key
- RSA Private Key

> - PKCS#8 Private Key

Reviewer 2: Do you think adding regex patterns here makes sense?

Author: Not sure, regex patterns are details of implementation and can be changed even to not regexp but strings manipulation. Also not sure do they provide any value for the end customer
